bin/hdfcoinc/pycbc_multiifo_coinc_statmap

```diff
@@ -109,37 +102,40 @@ else:
 
 logging.info("We have %s triggers" % len(all_trigs.stat))
 fore_locs = all_trigs.timeslide_id == 0
-# Foreground trigger times for ifo.
+
+# Foreground trigger times for ifos
 fore_time = {}
 for ifo in ifos:
     fore_time[ifo] = all_trigs.data['%s/time' % ifo][fore_locs]
-# Average times of triggers from ifo1 and ifo2
-ave_fore_time = 0
-for ifo in ifos:
-    ave_fore_time += fore_time[ifo] / len(ifos)
-
-# Remove start and end time around every average foreground trigger time to
-# window around.
+# Average times of triggers (note that coincs where not all ifos have triggers
+# will contain -1 sentinel values)
+fore_time_zip = zip(*fore_time.values())
+ave_fore_time = []
+for ts in fore_time_zip:
+    ave_fore_time.append(coinc.mean_if_greater_than_zero(ts)[0])
+ave_fore_time = numpy.array(ave_fore_time)
+
+# Window out start and end time around every average foreground trigger time
 remove_start_time = ave_fore_time - args.veto_window
 remove_end_time = ave_fore_time + args.veto_window
 
-# The time contained between segments around the times contained between each
-# element of remove_start_time and remove_end_time.
+# Total amount of time removed around foreground triggers
 veto_time = abs(veto.start_end_to_segments(remove_start_time,
                                            remove_end_time).coalesce())
 
-# Veto indices from list of triggers for times in ifo 1&2 around the window
-# times. This gives exclusive background triggers.
-veto_indeces = {}
+# Veto indices from list of triggers in the windowed times around fg triggers
+# This gives exclusive background triggers
+exc_zero_trigs = all_trigs.remove([])  # Start by copying existing triggers
 for ifo in ifos:
-    veto_indeces[ifo] = veto.indices_within_times(all_trigs.data['%s/time' % ifo], remove_start_time,
-                                                  remove_end_time)
-    exc_zero_trigs = all_trigs.remove(veto_indeces[ifo])
+    fg_veto_ids = veto.indices_within_times(
+                       exc_zero_trigs.data['%s/time' % ifo],
+                       remove_start_time, remove_end_time)
+    exc_zero_trigs = exc_zero_trigs.remove(fg_veto_ids)
```

Comment:
Was this the main bug? I.E. it was always starting out with the all_trigs and thus wouldn't be cumulatively removing?

Reply:
well also the wrong time calc for doubles in triple time, but this all_trigs.remove is the main bug, yes.